imports/plugins/included/product-detail-simple/client/containers/productDetailContainer.js

```diff
@@ -56,12 +58,38 @@ class ProductDetailContainer extends Component {
         return [];
       }
 
+      if (this.props.storedCart.items && this.props.storedCart.items) {
+        this.props.storedCart.items.forEach((item) => {
+          if (item.variants._id === currentVariant._id) {
+            storedQuantity = item.quantity;
+          }
+        });
+      }
+
       quantity = parseInt(this.state.cartQuantity, 10);
+      totalQuantity = quantity + storedQuantity;
 
       if (quantity < 1) {
         quantity = 1;
       }
 
+      if (currentVariant.inventoryPolicy && quantity > currentVariant.inventoryQuantity && storedQuantity < currentVariant.inventoryQuantity) {
+        Alerts.inline("Your product quantity has been adjusted to the max quantity in stock", "warning", {
+          placement: "productDetail",
+          autoHide: 10000
+        });
+        quantity = currentVariant.inventoryQuantity;
+        totalQuantity = totalQuantity - currentVariant.inventoryQuantity;
+      }
+
+      if (currentVariant.inventoryPolicy && totalQuantity > currentVariant.inventoryQuantity) {
+        Alerts.inline(`Sorry, cart contains maximum quantity of ${currentVariant.title}. Failed to add to cart.`, "error", {
```

Comment:
Also the message is confusing. How does my cart contain the maximum quantity? We are denying because it's out of stock right? The message should reflect that somehow.